### ISSUE TITLE
Code Grabber Crashing (again)

### ISSUE THREAD

**Issue by terininetails** [Lumbridge/SmiteMixerCodeRedeemer#22]
I had this issue come up sometimes with 1.0.6, but for 1.0.7 and 1.0.7.1 it's happening almost every time a code is detected. "Smite Mixer Code Grabber.exe has stopped working," with no further details available. Currently running on Windows 10 64-bit, with 32-bit Smite. It will crash regardless of if AFK mode is checked or not, and I do not have any other boxes checked (whitelist, minimize, notifications, or sounds). I'm running as an Administrator. 

Also, sometimes when I attempt to manually redeem a code, I get an error with the attached. 
[1071errordump.txt](https://github.com/Lumbridge/SmiteMixerCodeRedeemer/files/1895833/1071errordump.txt)

**Comment by Lumbridge** [comment]
Can I assume you’ve tried deleting all files (including .dlls) and downloading latest version fresh?

**Comment by terininetails** [comment]
Indeed I am - I make sure to delete all the files before extracting the new ones.

**Comment by Lumbridge** [comment]
Strange, I haven’t had any other reports of this. It could be due to residual files in a temp directory. As for the clipboard error dump, I will fix this.. it’s a limitation of a method being used which will be replaced with a better one. Also please see issue #21 where @kaimi discusses running in admin mode.
Cheers for the info :)

**Comment by terininetails** [comment]
Aha, I _**do**_ run Smite as an Administrator. I'll stop doing that and see if that helps.

Thanks!

**Comment by terininetails** [comment]
Alright, update on my weird situation. I do not run either Smite or the Code Grabber as an Admin, anymore. Then I watched as the first code was grabbed - it was a fake code, but I don't think that would have any effect on this outcome. I'm trying to be as concise as possible, having had to debug things from the other side, before. :) 

**First situation:** Code grabber was running first, then I brought up Smite and put it in the foreground, set to AFK mode with no other options checked. As soon as a code was grabbed, it gave the generic "stopped working" error. 

**Second situation:** Restarted the code grabber while Smite was running, and did not set to AFK mode (all boxes were unchecked). The code was grabbed, but did not crash. 

I clicked "Redeem Selected Code" and got the "clipboard error", but I could paste into the chat box and it came up with the entire "/claimpromotion AP***************" string. Clicking "Continue" on the clipboard error simply dismissed it. 

The code did not switch to "Redeemed: True" in the code grabber. I attempted to again manually redeem the code, and it went through normally, pasting the "/claimpromotion AP***************" string in the Smite chat box, and then switching the redeemed status to "True".

As an experiment, I clicked the other two buttons afterward:

"Redeem All Active Codes" did nothing, and nothing was copied to the clipboard, but that was possibly because the code was flagged with "Redeemed: True". It didn't crash the program or give a clipboard error.

"Copy Selected Code to Clipboard" functioned normally, without crashing the program or giving a clipboard error.

**Comment by Lumbridge** [comment]
Alright then, interesting. Upon investigation it seems that the Clipboard is quite a temperamental thing to program with - I'll try make sure this is fixed ASAP.

**Comment by terininetails** [comment]
**UPDATE:** My "first time with a given code" theory just got shot down. A new code just appeared while in AFK mode, and it got redeemed properly.

**Comment by Lumbridge** [comment]
This is expected as the Clipboard memory area is shared among all active processes so errors can appear quite often, I'll make sure it's handled correctly soon.

**Comment by terininetails** [comment]
> FWIW, it works for me no matter in what order Smite and the Grabber are started.

I didn't think the order of Smite/Grabber being launched would have much to do with anything, but I included the information just in case.

**Comment by Lumbridge** [comment]
Any crashes @kaimi ?

**Comment by kaimi** [comment]
> I didn't think the order of Smite/Grabber being launched would have much to do with anything

Me neither, just trying to help rule that out.

> Any crashes @kaimi ?

Nope, none.

**Comment by Lumbridge** [comment]
Closed due to inactivity.

**Comment by terininetails** [comment]
An update on this, since a pretty big development has happened -

I got an entirely new computer. New hard drive, new installations of everything, the whole bit. I STILL got this issue with the Code Redeemer crashing.

But it seems I've found a workaround, for now - if I click to copy a code to the clipboard (I have to debug a new active code to do this), and THEN try to use the "Redeem" button, it works perfectly. Doing these steps before setting AFK mode makes everything work normally.

Hopefully this information helps in your work!
